File: server/GameServer.py
from twisted.internet.protocol import DatagramProtocol
from twisted.internet import reactor

# Server configuration
HOST = '0.0.0.0'
PORT = 6793

class GameServerProtocol(DatagramProtocol):
    def startProtocol(self):
        print(f'Serving on {HOST}:{PORT}')

    def datagramReceived(self, data, addr):
        print(f"Received message from {addr}: {data}")

        # Process the message and perform any game-related logic here
        # Update game state, respond to the client, etc.

        # Do not send a response back to the client, just print to server console

def main():
    reactor.listenUDP(PORT, GameServerProtocol(), interface=HOST)
    reactor.run()

if __name__ == '__main__':
    main()

File: servers/GameServer.ts
// Server configuration
const HOST = '0.0.0.0';
const PORT = 6739;

class GameServer {
  private udpServer: Deno.DatagramConn | null = null;

  async start(): Promise<void> {
    try {
      console.log(`[Game Server] Online! ${HOST}:${PORT}`);
      this.udpServer = await Deno.listenDatagram({ hostname: HOST, port: PORT, transport: "udp" });

      for await (const [data, addr] of this.udpServer) {
        this.datagramReceived(data, addr);
      }
    } catch (error) {
      console.error(`[Game Server] Error: ${error}`);
    }
  }

  datagramReceived(data: Uint8Array, addr: Deno.Addr): void {
    const decoder = new TextDecoder();
    const message = decoder.decode(data);

    // Extract port number from string representation of addr
    const portMatch = addr.toString().match(/:(\d+)$/);
    const port = portMatch ? portMatch[1] : "unknown";

    console.log(`[Game Server] Received message from ${addr}:${port}: ${message}`);

    // Process the message and perform any game-related logic here
    // Update game state, respond to the client, etc.

    // Do not send a response back to the client, just print to server console
  }
}

async function main() {
  const server = new GameServer();
  await server.start();
}

if (import.meta.main) {
  main();
}
